pulsar-client/src/main/java/org/apache/pulsar/client/impl/ConsumerImpl.java

```diff
@@ -1855,10 +1863,20 @@ private void initDeadLetterProducerIfNeeded() {
             createProducerLock.writeLock().lock();
             try {
                 if (deadLetterProducer == null) {
-                    deadLetterProducer = client.newProducer(Schema.AUTO_PRODUCE_BYTES(schema))
-                            .topic(this.deadLetterPolicy.getDeadLetterTopic())
-                            .blockIfQueueFull(false)
-                            .createAsync();
+                    // We first need to create the initial subscription for this DLQ topic.
+                    // Otherwise, when we do not set the retention, it may lead to data loss.
+                    // The default initial subscription name is the subscription name of the current consumer.
+                    CompletableFuture<Consumer<byte[]>> deadLetterConsumer =
+                            client.newConsumer(Schema.AUTO_PRODUCE_BYTES(schema))
```

Comment:
If the subscription is already there, we shouldn't try to create the consumer, otherwise we might get a `ConsumerBusy` error.

Reply:
`AUTO_PRODUCE_BYTES` is meant to be used when producing, but not in a consumer.